src/modules/launcher/PowerLauncher/ViewModel/MainViewModel.cs

```diff
@@ -440,54 +441,56 @@ private void QueryResults()
                 var currentCancellationToken = _updateSource.Token;
                 _updateToken = currentCancellationToken;
 
-                ProgressBarVisibility = Visibility.Hidden;
                 var query = QueryBuilder.Build(QueryText.Trim(), PluginManager.NonGlobalPlugins);
                 if (query != null)
                 {
-                    // handle the exclusiveness of plugin using action keyword
-                    RemoveOldQueryResults(query);
-
                     _lastQuery = query;
-                    var plugins = PluginManager.ValidPluginsForQuery(query);
-
                     Task.Run(() =>
                     {
-                        // so looping will stop once it was cancelled
-                        var parallelOptions = new ParallelOptions { CancellationToken = currentCancellationToken };
+                        Thread.Sleep(20);
```

Comment:
What is the sleep for, and can we better syncronize this?

Reply:
Without thread.sleep I noticed quirkiness as a lot of queries were going through and ending up updating list view.